index.js

```diff
@@ -149,42 +180,60 @@ const line = []
 
 
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 1", "color: red")
+console.log("%cScope & Closures - Question 1", "color: red")
+
+let ticketNumber = 0
+function takeATicketNumber(line) {
+  ticketNumber++
+  line.push(ticketNumber)
+  return `Welcome. You are ticket number ${ticketNumber}`
+}
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 1`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 1`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 2`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 2`
 
-// console.log(nowServing(line))
-// // => `Currently serving 1.`
+console.log(nowServing(line))
+// => `Currently serving 1.`
 
-// console.log(nowServing(line))
-// // => `Currently serving 2.`
-// console.log("%c----------", "color: red") 
+console.log(nowServing(line))
+// => `Currently serving 2.`
+console.log("%c----------", "color: red") 
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 3`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 3`
 
 
 // ***** Scope & Closures - Question 2 *****
 
 
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 2", "color: red")
+console.log("%cScope & Closures - Question 2", "color: red")
+
+const newLine = []
+
+function ticketNumberGeneratorFunc(arr) {
+  ticketNumber = 0
+  function takeATicketNumber (arr) {
+    ticketNumber++
+    arr.push(ticketNumber)
+    return `Welcome. You are ticket number ${ticketNumber}`
+  }
+  return takeATicketNumber(arr)
+}
```

Comment:
Nice work on this, there are just a couple things to check in your code. 

Be careful on line 218 here, if you declare a variable without `let` or `const` it will become a global variable (even if you declare it inside a function). To make sure that the ticket variable here is scoped to the function, use `let`.

Also, on line 224, you should return a reference to the takeATicketNumber function **without** invoking it:

```js
return takeATicketNumber
```

If you invoke the function, you're no longer returning a function, you're returning a string (since that's what `takeATicketNumber` returns if you invoke it).